src/vs/editor/contrib/codeAction/browser/codeActionMenu.ts

```diff
@@ -608,12 +656,45 @@ export class CodeActionMenu extends Disposable implements IEditorContribution {
 		});
 	}
 
+	private showContextViewHelper(menu: CodeActionMenu, params: ICodeActionMenuParameters, menuActions: IAction[]) {
+		params.menuObj._contextViewService.showContextView({
+			getAnchor: () => params.anchor,
+			render: (container: HTMLElement) => params.menuObj.renderCodeActionMenuList(container, menuActions, params),
+			onHide: (didCancel: boolean) => {
+				const openedFromString = (params.options.fromLightbulb) ? CodeActionTriggerSource.Lightbulb : params.trigger.triggerAction;
+				params.menuObj.codeActionTelemetry(openedFromString, didCancel, params.codeActions);
+				params.menuObj._visible = false;
+				params.menuObj._editor.focus();
+			},
+		},
+			params.menuObj._editor.getDomNode()!, false,
+		);
+
+	}
+
+	public static toggleDisabledOptions(params: ICodeActionMenuParameters): void {
+		params.menuObj.hideCodeActionWidget();
+
+		showDisabled = !showDisabled;
+
+		const actionsToShow = showDisabled ? params.codeActions.allActions : params.codeActions.validActions;
+
+		const menuActions = params.menuObj.getMenuActions(params.trigger, actionsToShow, params.codeActions.documentation);
+
+		params.menuObj.showContextViewHelper(params.menuObj, params, menuActions);
+	}
+
 	public async show(trigger: CodeActionTrigger, codeActions: CodeActionSet, at: IAnchor | IPosition, options: CodeActionShowOptions): Promise<void> {
 		const model = this._editor.getModel();
 		if (!model) {
 			return;
 		}
-		const actionsToShow = options.includeDisabledActions ? codeActions.allActions : codeActions.validActions;
+
+		let actionsToShow = options.includeDisabledActions ? codeActions.allActions : codeActions.validActions;
+
+		if (trigger.triggerAction === CodeActionTriggerSource.Refactor) {
```

Comment:
I think `includeDisabledActions` should be the only thing the controls if disabled actions are added. Also checking if the source was `Refactor` does not seem correct

Reply:
In the event of a cancelled or accepted code action, it runs through the `show` function again, and there I have to check if the `showDisabled` has been modified so that it syncs with `includeDisabledActions`, specifically only for refactorings, not quickfixes, source actions, or lightbulb widget activations.